### PR TITLE
chore: return error in arrow array convert function

### PR DESCRIPTION
**Pull request by QuenKar**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

If there is a cast error in `try_from_arrow_timestamp_array` and `try_from_arrow_time_array` function, it returns `CastTypeSnafu` now.

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [x]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link (optional)
Closes https://github.com/GreptimeTeam/greptimedb/issues/2010

### CODE REVIEW

File: src/datatypes/src/vectors/primitive.rs [hunk 1]
```diff
@@ -142,8 +148,11 @@ impl<T: LogicalPrimitiveType> PrimitiveVector<T> {
                     .to_data(),
                 _ => unreachable!(),
```
Comment (evenyag) [inline src/datatypes/src/vectors/primitive.rs:149]: question: Not sure whether other units are possible to use. Is it really unreachable? @v0y4g3r

Comment (QuenKar) [reply src/datatypes/src/vectors/primitive.rs:149]: @evenyag May be I can answer this question. 
In my initial version,I modify all `unreachable!()` to `CastTypeSnafu`, but when I want to test this logic branch,
I find that `arrow-array` only support `Time32SecondArray`,`Time32MillisecondArray`,`Time64MicrosecondArray` and `Time64NanosecondArray`.  that means we can't get like `Time32MicrosecondArray` or `Time64SecondArray` and so on. So I think it's unreachable.

**Review by killme2008 — APPROVED** [review]
LGTM

**Review by evenyag — APPROVED** [review]
LGTM

**Pull request merged**